### DISCUSSION TITLE
Playing Sd card samples

### DISCUSSION THREAD

**Discussion by souravlal** [marcel-licence/esp32_midi_sampler#33] [Q&A]
Is there any modifications needed on the code or hardware side to play samples from sd card? Or can I just put an sd card to the esp32 audio kit and it will automatically load samples?

**Comment by Slider2732** [comment]
I only use the SD card, with 99 samples on it. 16 bit, 44.1K.
A pointer for practical use of a larger sample library is to move often used samples further up the directory listing.   
Loading is sequential, not a type of random access, so it makes sense to put samples you'll find most use out of at the start of file namings. Example, if you'd like 'VoxHumana.wav' to show up first in the SD card directory, you might rename it to '1VoxHumana.wav' because a '1' will be listed far higher than a 'V' on the SD card. Then, '2RichChoir.wav', '3LatelyBass.wav', '4RhodesPiano.wav' etc. 

Changes can be made in esp32_midi_sampler.ino, such that it will load an SD card file at boot up. 

    /*
    * use this to activate auto loading
    */
    //#define AUTO_LOAD_PATCHES_FROM_LITTLEFS
    #define AUTO_LOAD_PATCHES_FROM_SD_MMC

Then, a little further down, the name of the file you want to auto load, in my case 1.wav

    /* select sd card */
    #ifdef AUTO_LOAD_PATCHES_FROM_SD_MMC
    PatchManager_SetDestination(1, 1);
    Sampler_LoadPatchFile("/samples/1.wav");
    //Sampler_LoadPatchFile("/samples/76_Pure.wav");
    #endif


If you're running the new code release with screen support, you can then load whatever you like from the SD card.

**Reply by souravlal** [reply]
What if I don't use a display? Will I be able to load whatever samples I want?
Does the sample files are need to be inside a folder?

**Reply by marcel-licence** [reply]
You can load samples at startup with the code @Slider2732  has shown above. You can go without a display. You can also use TeraTerm, enable VT100 in the config.h and use your computer to show some information (top right corner): https://youtu.be/7uSobNW7_A4

**Comment by Slider2732** [comment]
I'd advise the display, it makes it a bunch easier to know what's going on. 
With the old code, I changed quite a bit around and had it load sequentially or be able to skip to the next bank of sounds... my samples were in 10's on the SD card based on their type (piano, bass, strings, organs etc). All of that was only temporary, the new release of Marcel's code increased functionality a huge amount. 
All I know at the moment, is to press Key2 a couple of times, then Key1 to step through the samples and Key4 to load it. You'd do that with no screen, which isn't the easiest way.
The menu system is still a bit confusing, but I expect it's due to lack of familiarity. 
However, if you get an Arduino to output CC messages then you can sequentially load samples using buttons on the Arduino. One button to step to the next sample, another to load it.   

Folder is 'samples' in the root of the card. The card has to be formatted to regular Microsoft FAT32.

**Reply by souravlal** [reply]
Thank you guys for your replies.
